manifests/package.pp

```diff
@@ -1,32 +1,53 @@
-# == Define: types::package
+# @summary Define for package types
+#   For a full description of the exec type check: https://www.puppet.com/docs/puppet/7/types/package.html
+#
+# @param ensure
+#   Value passed to ensure attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-ensure for more information.
+#
+# @param adminfile
+#   Value passed to adminfile attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-adminfile for more information.
+#
+# @param allowcdrom
+#   Value passed to allowcdrom attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-allowcdrom for more information.
+#
+# @param configfiles
+#   Value passed to configfiles attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-configfiles for more information.
+#
+# @param install_options
+#   Value passed to install_options attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-install_options for more information.
+#
+# @param provider
+#   Value passed to provider attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-provider for more information.
+#
+# @param responsefile
+#   Value passed to responsefile attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-responsefile for more information.
+#
+# @param source
+#   Value passed to source attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-source for more information.
+#
+# @param uninstall_options
+#   Value passed to uninstall_options attribute. Unused if empty.
+#   Check https://www.puppet.com/docs/puppet/7/types/package.html#package-attribute-uninstall_options for more information.
 #
 define types::package (
-  $ensure            = 'present',
-  $adminfile         = undef,
-  $allowcdrom        = undef,
-  $configfiles       = undef,
-  $install_options   = undef,
-  $provider          = undef,
-  $responsefile      = undef,
-  $source            = undef,
-  $uninstall_options = undef,
+  Enum['present', 'absent', 'purged', 'disabled', 'installed', 'latest'] $ensure            = 'present',
```

Comment:
Any reason why not Stdlib::Ensure::Package is not used here?

```
Validate the value of the ensure parameter for a package

Alias of Variant[Enum['present', 'absent', 'purged', 'disabled', 'installed', 'latest'], String[1]]```

Reply:
`Stdlib::Ensure::Package` isn't available in stdlib 5.0.0 that is the current minimum version for this module. Should I change it in our fork ?

Reply:
Let's leave it for now.

Reply:
see #2